libobs/obs-output.c

```diff
@@ -798,17 +819,19 @@ void obs_output_remove_encoder(struct obs_output *output,
 	if (!obs_output_valid(output, "obs_output_remove_encoder"))
 		return;
 
-	if (output->video_encoder == encoder) {
-		output->video_encoder = NULL;
-	} else {
-		for (size_t i = 0; i < MAX_AUDIO_MIXES; i++) {
-			if (output->audio_encoders[i] == encoder)
-				output->audio_encoders[i] = NULL;
-		}
+	for (size_t i = 0; i < MAX_VIDEO_ENCODERS; i++) {
+		if (output->video_encoders[i] == encoder)
+			output->video_encoders[i] = NULL;
+	}
+
+	for (size_t i = 0; i < MAX_AUDIO_MIXES; i++) {
+		if (output->audio_encoders[i] == encoder)
+			output->audio_encoders[i] = NULL;
 	}
 }
 
-void obs_output_set_video_encoder(obs_output_t *output, obs_encoder_t *encoder)
+void obs_output_set_video_encoder2(obs_output_t *output, obs_encoder_t *encoder,
+				   size_t idx)
 {
 	if (!obs_output_valid(output, "obs_output_set_video_encoder"))
```

Comment:
This should be renamed to `obs_output_set_video_encoder2`. It's only for logging, but other functions have their "v2" functions with their own valid checks too.

Reply:
Ya, just missed that. Will fix